lib/diagnostic.rb

```diff
@@ -27,8 +27,17 @@ def example
 # responds successfully and lists all examples.
```

Comment:
Q1:

# BDD is meant to test for expected behavior of parts of an application.
# Tests are meant to be written before application code, and the tests will
# drive development towards working code. This differs from TDD in that
# testing for behavior is more meaningful than arbitrary tests and also
# BDD tends to read more easily than TDD, making the tests all around more
# impactful.